src/dependencies.rs

```diff
@@ -199,3 +203,85 @@ fn create_dependencies(
         alert_dao,
     }
 }
+#[cfg(test)]
+mod tests {
```

Comment:
Can we add more tests, in particular of generic parts like checking that setting pragma works, that migrating to an older version fails, that migration to the same version does nothing, etc...

Reply:
Add more granular tests - ok